app/lib/actions/katello/host/hypervisors_update.rb

```diff
@@ -10,17 +10,20 @@ class HypervisorsUpdate < Actions::EntryAction
 
         def run
           output[:results] = input[:results]
-        end
-
-        def finalize
           @hypervisors = input[:hypervisors]
           return unless @hypervisors
 
           User.as_anonymous_admin do
-            load_resources
+            ActiveRecord::Base.transaction do
+              load_resources
+
+              @hosts.each do |uuid, host|
+                update_subscription_facet(uuid, host)
+              end
+            end
```

Comment:
out of curiosity, why did you move this to the run phase and put it in a transaction ?

Reply:
The FactImporter needs to run outside of a transaction, so I moved the code to the run phase. But I also wanted the rest of the code to run in transaction, so I wrapped it with one. Should I add a comment with this explanation?